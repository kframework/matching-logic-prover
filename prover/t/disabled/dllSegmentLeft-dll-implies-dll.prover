claim \implies( \and( dllSegmentLeft ( 
                            H { ArrayIntInt }
                          , X { Int }
                          , Y { Int }
                          , F { SetInt }
                          )
                    , dll ( H { ArrayIntInt }
                          , Y { Int }
                          , G { SetInt }
                          )
                    , \equals( K { SetInt }
                             , union( F { SetInt }
                                    , G { SetInt }
                                    )
                             )
                    , disjoint( F { SetInt }
                              , G { SetInt }
                              )
                    )
              , \and( dll ( H { ArrayIntInt }
                          , X { Int }
                          , K { SetInt }
                          )
                    )
              )
// strategy search-bound(3)
strategy simplify  ; kt ; normalize 
       ; ( (simplify ; instantiate-existentials ; smt)
         | (simplify ; right-unfold-Nth(0,1) ; simplify ; instantiate-existentials ; wait)
         | (wait)
         )
