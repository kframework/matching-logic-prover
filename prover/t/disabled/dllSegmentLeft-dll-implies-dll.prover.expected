<prover>
  <goal>
    <id>
      root
    </id>
    <active>
      true
    </active>
    <parent>
      .
    </parent>
    <k>
      \implies ( \and ( dllSegmentLeft ( H { ArrayIntInt } , X { Int } , Y { Int } , F { SetInt } , .Patterns ) , dll ( H { ArrayIntInt } , Y { Int } , G { SetInt } , .Patterns ) , \equals ( K { SetInt } , union ( F { SetInt } , G { SetInt } , .Patterns ) ) , disjoint ( F { SetInt } , G { SetInt } , .Patterns ) , .Patterns ) , \exists { .Patterns } \and ( dll ( H { ArrayIntInt } , X { Int } , K { SetInt } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
